WalletWasabi/Crypto/ZeroKnowledge/Credential.cs

```diff
@@ -0,0 +1,22 @@
+﻿using NBitcoin.Secp256k1;
+using WalletWasabi.Crypto.Groups;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public class Credential
+	{
+		public Credential(Scalar amount, Scalar randomness, MAC mac)
+		{
+			Amount = amount;
+			Randomness = randomness;
+			Mac = mac;
+		}
+
+		public Scalar Amount { get; } // could consider uint, but only really makes sense for credential requests because of range proofs
+		public Scalar Randomness { get; }
+		public MAC Mac { get; }
+
+		public CredentialPresentation Present(Scalar z)
+			=> CredentialPresentation.FromMAC(Mac, z, Amount, Randomness);
```

Comment:
the only thing i can think of is maybe it makes sense to just move `FromMAC`'s body here and eliminate it, since it makes more sense for `Credential` to knowhow to randomize itself than for `CredentialPresentation` to know about MAC and and how the amount commitment and serial number are built